python/rascaline/rascaline/utils/splines.py

```diff
@@ -401,3 +448,195 @@ def _radial_integral_derivative(
             return super()._radial_integral_derivative(n, ell, positions)
         else:
             return self.radial_integral_derivative_funcion(n, ell, positions)
+
+
+class RealSpaceSpliner(RadialIntegralSplinerBase):
+    def __init__(
+        self,
+        cutoff: float,
+        max_radial: int,
+        max_angular: int,
+        basis: RadialBasisBase,
+        density: AtomicDensityBase,
+        accuracy: float = 1e-8,
+    ):
+        if not HAS_SCIPY:
+            raise ValueError("Spliner class requires scipy!")
+
+        self.density = density
+
+        super().__init__(
+            max_radial=max_radial,
+            max_angular=max_angular,
+            spline_cutoff=cutoff,
+            basis=basis,
+            accuracy=accuracy,
+        )
+
+    def _radial_integral(self, n: int, ell: int, positions: np.ndarray) -> np.ndarray:
+        # Use analytical integrals if available
+        if type(self.density) is DeltaDensity:
+            return self._radial_integral_delta(n, ell, positions, derivative=False)
+        elif type(self.density) is GaussianDensity:
+            return self._radial_integral_gaussian(n, ell, positions, derivative=False)
+        else:
+            return self._radial_integral_custom(n, ell, positions, derivative=False)
```

Comment:
One could factor `_radial_integral_delta` and `_radial_integral_gaussian` out, but I would keep this logic and the implementation here since it combines the atomic densities and radial basis.